examples/C++/src/demo5_line_search.cxx

```diff
@@ -0,0 +1,348 @@
+/*!
+  \file
+  \ingroup examples
+  \brief A class to compute a line search evaluation in the direction of
+  the gradient of a given image and objective function.
+  All parameters can be parsed from a parameter file. See `demo5_line_search.par`.
+
+  Give an image and objective function configuration, this script will perform a line search
+  from an minimum to an maximum step size (alpha).
+  Options are included to perform this line search linearly or using exponential step
+  size iterations.
+  Additionally, a lower positivity bound is applied to all computed image.
```

Comment:
```suggestion
```suggestion
  from a minimum to a maximum step size (alpha).
  Options are included to perform this line search linearly or using exponential step
  sizes.
  Additionally, a lower positivity bound is applied to all computed images.
```